Source/WebKit/WebProcess/WebPage/WebCookieJar.h

```diff
@@ -67,6 +76,7 @@ class WebCookieJar final : public WebCore::CookieJar {
     bool isEligibleForCache(WebFrame&, const URL& firstPartyForCookies, const URL& resourceURL) const;
 
     mutable WebCookieCache m_cache;
+    HashMap<String, WeakPtr<WebCore::CookieChangeListener>> m_cookieChangeListeners;
```

Comment:
This should be a `HashMap<String, WeakHashSet<WebCore::CookieChangeListener>>` since there can be more than one listener per host.